src/CprSniffer.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+/*
+ * This file is part of itk-dev/cpr-sniffer.
+ * (c) 2020 ITK Development
+ * This source file is subject to the MIT license.
+ */
+
+namespace ItkDev\CprSniffer;
+
+/**
+ * Class CprSniffer.
+ *
+ * Check a string for CPR resemblance.
+ */
+class CprSniffer
+{
+    // Due to lacking capacity of valid modulo 11 numbers some years, there are
+    // valid CPR numbers that don't pass modulo 11 check. In those cases we just
+    // return true.
+    // https://cpr.dk/cpr-systemet/personnumre-uden-kontrolciffer-modulus-11-kontrol/
+    private $noModuloCheckNumbers = [
+        '010160',
+        '010164',
+        '010165',
+        '010166',
+        '010169',
+        '010170',
+        '010174',
+        '010180',
+        '010182',
+        '010184',
+        '010185',
+        '010186',
+        '010187',
+        '010188',
+        '010189',
+        '010190',
+        '010191',
+        '010192',
+    ];
+
+    /**
+     * Check if string contains valid CPR.
+     *
+     * @param $string
+     *   The string to check
+     *
+     * @return bool true if string contains a number that could be considered cpr
+     */
+    public function checkCpr($string)
+    {
+        if (empty($string)) {
+            return false;
+        }
+
+        // Remove spaces and dashes to prepare for a simpler regex.
+        // This way we check 3 formats 1234561234, 123456-1234 and 123456 1234.
+        $stringConcatenated = str_replace(['-', ' '], '', $string);
+
+        // Search the concatenated string for 10 digits in a row.
+        $numberFound = preg_match('/(^|\D)\d{10}($|\D)/', $stringConcatenated, $result);
+
+        if ($numberFound) {
+            $number = $result[0];
+
+            // Remove prefix from number.
+            if (!empty($result[1])) {
+                $number = substr($number, 1);
+            }
+
+            // Remove suffix from number.
+            if (!empty($result[2])) {
+                $number = substr($number, 0, -1);
+            }
+
+            // Prepare number for modulo 11 check.
+            $arr = $this->stringSplit($number);
+
+            if ($this->mod11Chk($arr, $number) && $this->dateChk($number)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Create an array from a string.
```

Comment:
Why not use https://www.php.net/manual/en/function.str-split.php

> Converts a string to an array.

Reply:
This was changed, and the function removed